java/build/src/main/java/sleeper/build/dependencydraw/DrawDependencyGraph.java

```diff
@@ -0,0 +1,197 @@
+        }
+        return Pair.of(allInEdges, allOutEdges);
+    }
+
+    public Pair<List<List<String>>, List<List<String>>> getEdgesFromName(List<String> selectedNodesList, Graph g) {
+        List<List<String>> allInEdges = new ArrayList<>();
+        List<List<String>> allOutEdges = new ArrayList<>();
+
+        for (int i = 0; i < selectedNodesList.size(); i++) {
+            allInEdges.add(new ArrayList<>(g.getInEdges(String.valueOf(selectedNodesList.get(i)))));
+            allOutEdges.add(new ArrayList<>(g.getOutEdges(String.valueOf(selectedNodesList.get(i)))));
+        }
+        return Pair.of(allInEdges, allOutEdges);
+    }
+
+    public void drawGraph(Pair<List<String>, List<List<String>>> graphData) {
+        List<String> nodeIDs = graphData.getFirst();
+        List<List<String>> edges = graphData.getSecond();
+        Graph<String, String> g = new DirectedSparseGraph<>();
+        for (String node : nodeIDs) {
+            g.addVertex(node);
+        }
+        for (List<String> edge : edges) {
+            g.addEdge(edge.get(0) + "---" + edge.get(1), edge.get(0), edge.get(1));
+        }
+        Layout<Integer, String> layout = new CircleLayout(g);
+        layout.setSize(new Dimension(900, 900));
+        VisualizationViewer<Integer, String> vv = new VisualizationViewer<Integer, String>(layout);
+        DefaultModalGraphMouse gm = new DefaultModalGraphMouse();
+        Function<String, Paint> edgePaint = s -> {
+            List<Collection<String>> edgesList = getEdges(vv, g).getFirst();
+            List<Collection<String>> edgedOutList = getEdges(vv, g).getSecond();
+            List<String> nextNodes = new ArrayList<>();
+            for (Collection<String> edgeIn : edgedOutList) {
+                for (String edge : edgeIn) {
+                    nextNodes.add(String.valueOf(edge).split("---")[1]);
+                }
+            }
+            List<List<String>> edgeNames = getEdgesFromName(nextNodes, g).getSecond();
+            for (Collection<String> edgeIn : edgesList) {
+                if (edgeIn.contains(s)) {
+                    return Color.RED;
+                } else {
+                    for (Collection<String> edgeOut : edgedOutList) {
+                        if (edgeOut.contains(s)) {
+                            return Color.BLUE;
+                        } else {
+                            if (showTransitiveDependencies) {
+                                for (int k = 0; k < edgeNames.size(); k++) {
+                                    List<String> nextOutEdge = edgeNames.get(k);
+                                    if (nextOutEdge.contains(s)) {
+                                        return Color.BLACK;
+                                    }
+                                }
+                            }
+                            return Color.lightGray;
+                        }
+                    }
+                }
+            }
+            return Color.BLACK;
+        };
```

Comment:
This function can be extracted to private members of the class and called in this main function. It would make this drawGraph method a bit easier to read.